api/custom_auth/oauth/serializers.py

```diff
@@ -65,20 +92,50 @@ def _get_user(self, user_data: dict):
             ):
                 raise PermissionDenied(USER_REGISTRATION_WITHOUT_INVITE_ERROR_MESSAGE)
 
-            return UserModel.objects.create(**user_data, sign_up_type=sign_up_type)
+            return UserModel.objects.create(
+                **user_data, email=email.lower(), sign_up_type=sign_up_type
+            )
+        elif existing_user.auth_type != self.get_auth_type().value:
+            # In this scenario, we're seeing a user that had previously
+            # authenticated with another authentication method and is now
+            # authenticating with a new OAuth provider.
+            user_model_id_attribute = self.get_user_model_id_attribute()
+            setattr(
+                existing_user,
+                user_model_id_attribute,
+                user_data[user_model_id_attribute],
+            )
+            existing_user.save()
 
         return existing_user
 
+    @abstractmethod
     def get_user_info(self):
         raise NotImplementedError("`get_user_info()` must be implemented.")
 
+    def get_auth_type(self) -> AuthType:
+        if not self.auth_type:  # pragma: no cover
+            raise NotImplementedError(
+                "`auth_type` must be set, or `get_auth_type()` must be implemented."
+            )
+        return self.auth_type
+
+    def get_user_model_id_attribute(self) -> str:
+        return self.user_model_id_attribute
```

Comment:
Yeah, true - I initially had it because it threw a NotImplementedError if the attribute was not set. But, to get the tests that test the base serializer class directly to pass, I had to set a default. 

I'll remove the method.

Reply:
Done